### PR TITLE
test(forms): raise coverage to 80%

### LINKED ISSUE #5935: Feature: Raise Codecov coverage for reinhardt-forms to 80%
## Prerequisites

- [x] I have searched existing issues to verify this coverage improvement has not already been requested.

## Summary

Raise Codecov line coverage for `reinhardt-forms` from **73.24%** to at least **80%** on the `main` branch.

Codecov reports the covered path `crates/reinhardt-forms/src` at 73.24%: 4,639 hit lines out of 6,334 instrumented lines (1,695 missed lines). At the current instrumented-line count, reaching 80% requires at least 5,068 hit lines, or 429 additional hit lines. The percentage target remains authoritative if the instrumented-line count changes.

The target is the smallest 10-point coverage boundary above the current 73.24% result.

## Motivation / Problem

The crate remains below the updated 80% coverage goal, leaving 1,695 instrumented lines without executable test coverage. The gap makes regressions in field parsing, validation boundaries, rendering variants, invalid input, and form submission error paths harder to detect and obscures whether the existing unit and integration suites exercise the supported behavior.

## Proposed Solution

- Add meaningful unit, integration, or compile-time consumer tests for field parsing, validation boundaries, rendering variants, invalid input, and form submission error paths.
- Prefer behavior-focused assertions for success, boundary, invalid-input, and failure cases instead of tests that only execute code.
- Prioritize uncovered supported behavior and observable failure paths; do not add tests whose only purpose is executing lines.
- Keep production changes limited to testability fixes or genuine defects discovered while adding coverage.
- Verify the focused tests locally and confirm the completed Codecov report reaches at least 80% for this crate.

## Acceptance Criteria

- The completed Codecov report for the implementation commit reports at least **80%** line coverage for `crates/reinhardt-forms/src`.
- Added tests contain meaningful behavioral assertions and comply with the repository testing standards.
- Relevant success, boundary, invalid-input, and failure cases are covered.
- Production behavior and public APIs remain backward compatible unless a genuine defect requires a separately justified fix.

## Alternatives Considered

- Treating the original 75% threshold as sufficient would leave additional supported behavior without regression protection.
- Excluding the crate or its untested paths from Codecov would hide the coverage gap rather than improve regression protection.
- Relying only on the workspace aggregate would allow this crate to remain below its explicit target.

## Is this a breaking change?

No - backward compatible

This issue is intended to add or improve tests and does not require public API changes.

## Migration Path (if breaking)

Not applicable.

## Additional Context

- Branch: `main`
- Baseline commit: [`7da062478448836ed0511cc8e924c1dbf6e33338`](https://github.com/kent8192/reinhardt-web/commit/7da062478448836ed0511cc8e924c1dbf6e33338)
- Codecov report: [baseline commit coverage tree](https://app.codecov.io/github/kent8192/reinhardt-web/commit/7da062478448836ed0511cc8e924c1dbf6e33338/tree/crates)
- The workspace total is 80.85%; this issue applies the target per crate.

### PR DESCRIPTION
**Pull request by kent8192**
## Summary

This PR raises reinhardt-forms line coverage from 73.24% to 80.43% with behavior-focused validation tests.

## Type of Change

- [x] Code quality improvements

## Breaking Change Assessment

- [x] **No** - This change is backward compatible

## Motivation and Context

Improve regression coverage for advanced fields and model-choice validation, including invalid password input and string primary-key choices.

Fixes #5935

## How Was This Tested?

- Focused tests: 2/2 passed
- Full package tests: 475/475 passed
- Target-aligned llvm-cov: 80.43% lines (5,434/6,756)
- rustfmt and diff checks passed

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have formatted the code with `cargo make fmt-fix`

## Labels to Apply

- [x] `code-quality` - Code quality improvements
- [x] `forms` - Form handling, validation, rendering

### Additional Context

The package clippy command still reports a pre-existing out-of-scope warning in `tests/field_validation_basic.rs`; no new forms-source warning was introduced.

### CODE REVIEW

File: crates/reinhardt-forms/src/fields/advanced_fields.rs [hunk 1]
```diff
@@ -985,10 +985,7 @@ impl FormField for PasswordField {
 		let s = match value.unwrap() {
 			Value::String(s) => s,
 			_ => {
-				return Err(FieldError::validation(
-					Some(&self.name),
-					"Invalid password format.",
-				));
+				return Err(FieldError::validation(None, "Invalid password format."));
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/advanced_fields.rs:992]: **Re: Classify the password behavior change as a fix**

The production error classification is already isolated in `fa154d3995` (`fix(forms): repair password and string key validation`), so release-plz will record the non-string password correction as a fix. No additional code change was needed.

`crates/reinhardt-forms/src/fields/advanced_fields.rs:L985-L989` - non-string password input returns the public validation error.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/fields/advanced_fields.rs [hunk 2]
```diff
@@ -1247,4 +1245,228 @@ mod tests {
...
+					Some(&json!("initial")),
+					Some("Identifier is optional."),
+					1,
+				),
+			),
+		);
+	}
+
+	#[rstest]
+	fn advanced_fields_cover_optional_invalid_type_and_change_boundaries() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/advanced_fields.rs]: The combined boundary coverage is now split into component-specific rstest cases in `crates/reinhardt-forms/src/fields/advanced_fields.rs:L1325-L1460`. Commit `6a704614e9` gives `UUIDField`, `DurationField`, `ColorField`, and `PasswordField` independent failure boundaries; the focused forms test suite passed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/fields/regex_field.rs [hunk 3]
```diff
@@ -519,4 +520,110 @@ mod tests {
 			);
 		}
 	}
+
+	#[test]
+	fn regex_slug_and_ip_fields_cover_configuration_and_boundary_contracts() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/regex_field.rs]: The combined field coverage is now split into independent rstest cases in `crates/reinhardt-forms/src/fields/regex_field.rs:L524-L636`. Commit `6a704614e9` isolates `RegexField`, `SlugField`, and `GenericIPAddressField`; the focused forms test suite passed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/form.rs [hunk 4]
```diff
@@ -1377,4 +1378,156 @@ mod tests {
 			&serde_json::json!("JOHN DOE")
 		);
 	}
+
+	#[test]
+	fn form_configuration_and_submission_edges_are_observable() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs:1564]: The `Form` unit coverage now ends before the multi-component bound-field scenario in `crates/reinhardt-forms/src/form.rs:L1382-L1517`. Commit `6a704614e9` moves the `Form`/`CharField`/`BoundField` interaction to `crates/reinhardt-forms/tests/form_bound_field_integration.rs:L6-L24`; both the crate tests and focused integration test passed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/fields/model_choice_field.rs [hunk 5]
```diff
@@ -665,4 +692,126 @@ mod tests {
...
+		);
+		assert_eq!(
+			ModelMultipleChoiceField::new("choices", Vec::<TestModel>::new())
+				.required(false)
+				.clean(Some(&json!("")))
+				.unwrap(),
+			json!([]),
+		);
+		assert!(!multiple.has_changed(Some(&json!(["1", "2"])), Some(&json!(["1", "2"]))));
+		assert!(multiple.has_changed(Some(&json!(["1", "2"])), Some(&json!(["2", "1"]))));
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/model_choice_field.rs]: The reordered-selection case now expects unchanged state, and `ModelMultipleChoiceField::has_changed` compares sorted normalized choice values in `crates/reinhardt-forms/src/fields/model_choice_field.rs:L507-L520`. Commit `e32da29c7c` also updates the regression assertion at `L787-L789`; the focused forms test suite passed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/fields/model_choice_field.rs [hunk 6]
```diff
@@ -512,7 +512,12 @@ impl<T: FormModel> FormField for ModelMultipleChoiceField<T> {
 				if a.len() != b.len() {
 					return true;
 				}
-				a.iter().zip(b.iter()).any(|(x, y)| x != y)
+
+				let mut initial_values: Vec<_> = a.iter().map(Value::to_string).collect();
+				let mut submitted_values: Vec<_> = b.iter().map(Value::to_string).collect();
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/model_choice_field.rs]: Fixed in commit 84815a6c60467a6bf824a8cce12289138035b300. choice_value_text now normalizes scalar JSON IDs to their textual values before sorting in crates/reinhardt-forms/src/fields/model_choice_field.rs:523-527, so numeric 1 and submitted string 1 compare equally; the regression assertion is at lines 812-815. Validation: cargo test -p reinhardt-forms --lib --all-features model_choice.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/tests/form_bound_field_integration.rs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+use std::collections::HashMap;
+
+#[rstest]
+fn form_and_bound_field_integration_preserves_prefixed_invalid_values() {
+	// Arrange
+	let mut form = Form::with_prefix("profile".to_string());
+	form.add_field(Box::new(
+		CharField::new("display_name".to_string()).required(),
+	));
+	form.bind(HashMap::from([("display_name".to_string(), json!(7))]));
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/tests/form_bound_field_integration.rs]: Fixed in commit 84815a6c60467a6bf824a8cce12289138035b300. Form now resolves the prefixed key through data_for_field for validation, change detection, and BoundField lookup in crates/reinhardt-forms/src/form.rs:866-870, while retaining the unprefixed fallback. The integration test binds profile-display_name at crates/reinhardt-forms/tests/form_bound_field_integration.rs:13-25. Validation: cargo test -p reinhardt-forms --test form_bound_field_integration --all-features.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/fields/model_choice_field.rs [hunk 8]
```diff
@@ -665,4 +697,126 @@ mod tests {
 			panic!("Expected array");
 		}
 	}
+
+	#[test]
+	fn model_choice_fields_validate_supported_input_shapes() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/model_choice_field.rs]: Fixed in commit 84815a6c60467a6bf824a8cce12289138035b300. The combined model-choice coverage is split into model_choice_field_validates_supported_input_shapes at crates/reinhardt-forms/src/fields/model_choice_field.rs:709-760 and model_multiple_choice_field_validates_supported_input_shapes at lines 762-836. Validation: cargo test -p reinhardt-forms --lib --all-features.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-forms/src/form.rs [hunk 9]
```diff
@@ -862,6 +862,13 @@ impl Form {
 			format!("{}-{}", self.prefix, field_name)
 		}
 	}
+
+	fn data_for_field(&self, field_name: &str) -> Option<&serde_json::Value> {
+		let prefixed_name = self.add_prefix_to_field_name(field_name);
+		self.data
+			.get(&prefixed_name)
+			.or_else(|| self.data.get(field_name))
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs]: Fixed in commit 9921814035. Form::data_for_field now reads an unprefixed key only when the form prefix is empty, so a prefixed form rejects an unrelated unprefixed submission. Added prefixed_forms_do_not_fallback_to_unprefixed_values; the full reinhardt-forms library suite passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 10]
```diff
@@ -226,7 +226,7 @@ impl Form {
 		}
 
 		for field in &self.fields {
-			let value = self.data.get(field.name());
+			let value = self.data_for_field(field.name());
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs:247]: Fixed in commit 9921814035. After a prefixed value is cleaned successfully, Form removes the submitted prefixed key before storing the canonical field name, so cleaned_data contains no profile-name entry. The full 373-test forms suite and form_bound_field_integration passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/fields/advanced_fields.rs [hunk 11]
```diff
@@ -1247,4 +1245,246 @@ mod tests {
 		let result = field.clean(Some(&json!("SecurePassword123")));
 		assert!(result.is_err());
 	}
+
+	#[rstest]
+	fn advanced_field_builders_expose_configured_metadata() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/advanced_fields.rs]: Fixed in commit a2554af8ae. The metadata coverage is now split into independent UUIDField, DurationField, and ComboField tests in crates/reinhardt-forms/src/fields/advanced_fields.rs. The focused builder tests and full forms library suite passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 12]
```diff
@@ -245,6 +245,8 @@ impl Form {
 							}
 						}
 					}
+					let submitted_name = self.add_prefix_to_field_name(field.name());
+					self.data.remove(&submitted_name);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs]: Fixed in commit `8a0288651fdf00e31c74dafbe3c601059447c77f`. `Form` now keeps submitted values in `data` and canonical values in `cleaned_data`; prefixed raw input remains available to `get_bound_field` and repeated `is_valid` calls. Added `prefixed_forms_preserve_bound_values_after_validation_failure` in `crates/reinhardt-forms/src/form.rs`. Validation: 374 forms unit tests and the BoundField integration test passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 13]
```diff
@@ -1350,6 +1362,42 @@ mod tests {
 		assert_eq!(form.add_prefix_to_field_name("email"), "user-email");
 	}
 
+	#[test]
+	fn prefixed_forms_do_not_fallback_to_unprefixed_values() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs]: Fixed in commit `8a0288651fdf00e31c74dafbe3c601059447c77f`. Added the `rstest` import and converted all three new prefixed-form cases in `crates/reinhardt-forms/src/form.rs` to `#[rstest]`. The forms unit tests passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 14]
```diff
@@ -1350,6 +1362,42 @@ mod tests {
...
+		let mut form = Form::with_prefix("profile".to_string());
+		form.add_field(Box::new(CharField::new("name".to_string()).required()));
+		form.bind(HashMap::from([(String::from("name"), json!("other-form"))]));
+
+		// Act
+		let valid = form.is_valid();
+
+		// Assert
+		assert!(!valid);
+		assert!(form.errors().contains_key("name"));
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs]: Fixed in commit `8a0288651fdf00e31c74dafbe3c601059447c77f`. The missing prefixed `name` case now asserts the exact error vector `['name']` in `crates/reinhardt-forms/src/form.rs`; the forms unit tests passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 15]
```diff
@@ -862,6 +864,15 @@ impl Form {
...
 		}
 	}
+
+	fn data_for_field(&self, field_name: &str) -> Option<&serde_json::Value> {
+		if self.prefix.is_empty() {
+			self.data.get(field_name)
+		} else {
+			let prefixed_name = self.add_prefix_to_field_name(field_name);
+			self.data.get(&prefixed_name)
+		}
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs:903]: Fixed in commit `8a0288651fdf00e31c74dafbe3c601059447c77f`. Documented the prefixed binding contract with public examples in `crates/reinhardt-forms/README.md` and `crates/reinhardt-forms/src/lib.rs`: submitted keys use the prefix, cleaned keys are canonical, and bound fields preserve submitted values for rerendering. `cargo doc -p reinhardt-forms --no-deps --all-features` passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/lib.rs [hunk 16]
```diff
@@ -44,6 +44,29 @@
 //! }
 //! ```
 //!
+//! ### Prefixed Form Data
+//!
+//! A prefixed form expects submitted field names to use the prefix. The
+//! validated values are exposed through canonical field names, while bound
+//! fields continue to read the original submitted values for rerendering.
+//!
+//! ```rust,ignore
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/lib.rs]: Changed the prefixed-form example in `crates/reinhardt-forms/src/lib.rs` from `rust,ignore` to an executable Rust doctest. Validation: `cargo test -p reinhardt-forms --doc --all-features` passed with 135 passed and the prefixed-form doctest at lib.rs:53 executed successfully. Commit: `d2e9897b82`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 17]
```diff
@@ -925,7 +973,14 @@ impl Form {
 	/// Returns a `BoundField` with the field's submitted data and errors attached.
 	pub fn get_bound_field<'a>(&'a self, name: &str) -> Option<BoundField<'a>> {
 		let field = self.get_field(name)?;
-		let data = self.data.get(name);
+		let data = if matches!(field.widget(), Widget::PasswordInput)
+			&& self.validation_complete
+			&& self.cleaned_field_names.contains(field.name())
+		{
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs]: Implemented sensitivity-based redaction in `crates/reinhardt-forms/src/form.rs:L973-L983`. `FormField::is_sensitive()` defaults to `false`, while `PasswordField` overrides it at `crates/reinhardt-forms/src/fields/advanced_fields.rs:L952-L958`; therefore successful password values remain redacted even when callers replace the public widget with `TextInput`. Regression coverage is in `crates/reinhardt-forms/tests/form_bound_field_integration.rs:L60-L75`.

Validation: `cargo test -p reinhardt-forms --all-features --test form_bound_field_integration` (4 passed).

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/fields/advanced_fields.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/fields/advanced_fields.rs:992]: Implemented the production correction as `d7f5e41803` (`fix(forms): redact sensitive bound values independently of widget`). The earlier non-string password correction remains in `fa154d3995` (`fix(forms): repair password and string key validation`). Test-only assertions are isolated in `1ec47a4564` (`test(forms): cover redaction with custom widgets and later errors`), so the user-visible changes are classified under Fixed.

Validation: `cargo test -p reinhardt-forms --all-features --test form_bound_field_integration` and `cargo test -p reinhardt-forms --all-features --lib has_changed_keeps_cleaned_values_after_later_field_error` passed.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-forms/src/form.rs [hunk 18]
```diff
@@ -1350,6 +1407,128 @@ mod tests {
...
+		});
+		form.bind(HashMap::from([("age".to_string(), json!("1"))]));
+
+		// Act
+		let valid = form.is_valid();
+
+		// Assert
+		assert!(!valid);
+		assert_eq!(form.cleaned_data().get("age"), Some(&json!(1)));
+		assert!(form.errors().contains_key("age"));
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-forms/src/form.rs]: Replaced the key-presence assertion with an exact comparison against `vec!["Age is not allowed."]` in `crates/reinhardt-forms/src/form.rs:L1481-L1484`. The regression now fails if the form-wide cleaner attaches a different error to `age`.

Validation: `cargo test -p reinhardt-forms --all-features --lib has_changed_keeps_cleaned_values_after_later_field_error` passed.

🤖 Generated with [Codex](https://openai.com/codex)
